docs/DeveloperGuide.md

```diff
@@ -154,6 +154,66 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Favourite feature
+
+#### Implementation
+
+The favourites feature is implemented by identifying the user's favourite contacts through their `favouriteStatus` state,  which is stored as a `Persons` field. The favourite-status of a person is represented by a `Favourite`, which can be either `IS_FAVOURITE`, or `NOT_FAVOURITE` singletons.
+
+Although it is a `Persons` field, it cannot be instantiated in an add command, nor modified through an edit command, unlike other fields such as "name" or "address". It should be thought of as metadata for the contact, and is set as `NOT_FAVOURITE` by default.
+
+As such, a contact's favourite-status is handled through the `FavouritesCommand`, which extends `Command`. Toggles the contact's favourite status based on their current favourite-status. Its execution behaves rather similarly to `edit`, and has its own `FavouritesCommand#createFavouritedPerson(target, newPerson)` operation.
+
+As part of the favourites feature, a user can also list all their favourite contacts through the `ListFavouritesCommand`, which extends `Command` as well. It leverages on the `PersonIsFavouriteContactPredicate`, which can be passed as an argument to `Model#updateFilteredPersonList(predicate)`
+
+Given below is an example scenario of the favourites feature in operation.
+
+Step 1. When a user is created, they are automatically assigned `NOT_FAVOURITE` as their favourite-status.
+
+![FavouriteState0](images/favourite/FavouriteState0.png)
+
+Step 2. The user executes `fav 3` command to favourite the 3rd person in the address book. The `fav` command executes and determines the current `favouriteStatus` of the contact (which in this case is `NOT_FAVOURITE`). It then calls the private helper method `createFavouritedPerson` to create a deep copy of the target contact, except with their `favouriteStatus` now pointing to the opposite `Favourite` value (which will be the `IS_FAVOURITE` singleton).
+
+![FavouriteState1](images/favourite/FavouriteState1.png)
+
+Step 3. The command execution continues on to replace the existing person object with the newly favourited contact through `Model#setPerson`. After the execution successfully terminates, the user can visually identify their favourite contacts through the yellow star right beside their contact names.
+
+![FavouriteState2](images/favourite/FavouriteState2.png)
+
+Step 4. The user decides that they want to see only their favourite contacts. The `favourites` command will then execute, passing the `PersonIsFavouriteContactPredicate` into the `Model#updateFilteredPersonList`.
+
+![FavouriteState3](images/favourite/FavouriteState3.png)
+
+The following sequence diagram shows how the favourite operation works:
+
+![FavouriteSequenceDiagram](images/favourite/FavouriteSequenceDiagram.png)
```

Comment:
I suggested this to the others as well, I think it would be clearer if you stated that PlantUML doesn't support cutting the lifeline after the destroy marker!